src/Mod/Path/Path/Op/Custom.py

```diff
@@ -47,21 +49,77 @@ def opFeatures(self, obj):
         return PathOp.FeatureTool | PathOp.FeatureCoolant
 
     def initOperation(self, obj):
+        obj.addProperty(
+            "App::PropertyEnumeration",
+            "Source",
+            "Path",
+            "Source of gcode (text, file, ...)"
+        )
+
+        obj.addProperty(
+            "App::PropertyFile",
+            "GcodeFile",
+            "Path",
+            "File containing gcode to be inserted",
+        )
+
         obj.addProperty(
             "App::PropertyStringList",
             "Gcode",
             "Path",
             QT_TRANSLATE_NOOP("App::Property", "The gcode to be inserted"),
         )
 
+        obj.Source = [ "Text", "File" ]
```

Comment:
I don't believe this is correct, it should be done through opPropertyEnumerations if I understand correctly.

obj.Source is the value, not the potential valid values and should probably default to "Text".

Reply:
I cannot remember which Op I was referencing when I wrote this, but it's done in several places. `Main/Job.py#525` for example. I'm looking into `opPropertyEnumberations` now, looks like I can pass this off to the Base class if used right.

Reply:
I've submitted PR #9165. If merged it will allow `opPropertyEnumerations` to be overridden on a subclass to return it's own enumeration values.

Reply:
#9165 has been merged.

Reply:
Great I'll update this PR and test to ensure it works as expected.